### PR TITLE
HT-3254: Bugfix: recluster commitments

### PR DESCRIPTION
**Pull request by jsteverman**
Commitments weren't being reclustered.

### CODE REVIEW

**Review by aelkiss — COMMENTED** [review]
Comments inline.

File: lib/clustering/reclusterer.rb [hunk 1]
```diff
@@ -53,5 +55,12 @@ def recluster_holdings
         .each {|batch| ClusterHolding.new(*batch.map(&:dup)).cluster.save }
     end
 
+    def recluster_commitments
```
[the remark this answers is not in the document]
Comment (mwarin) [reply lib/clustering/reclusterer.rb:58]: I have nothing to add here, but yeah just looking at this handful of methods I can see that we might want to refactor for DRY purposes.

**Review by mwarin — APPROVED** [review]
I like the tests, very neat.

File: spec/clustering/reclusterer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "cluster"
+require "clustering/cluster_commitment"
+require "clustering/cluster_ht_item"
+require "clustering/cluster_ocn_resolution"
+
+RSpec.describe Clustering::Reclusterer do
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply spec/clustering/reclusterer_spec.rb:9]: :+1:

**Review by aelkiss — APPROVED** [review]

File: lib/clustering/reclusterer.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsteverman) [reply lib/clustering/reclusterer.rb:58]: `recluster_batch` extracted with most recent commit.

"There's also the issue that we need to recluster in a certain order."  This is just a performance thing right? I think I am close to a better solution in HT-3237 (avoid reclustering when...) anyway. I'm constructing a graph of a cluster's OCNs, then performing a DFS to discover partitions. If I was smarter, something something disjoint-set might be better, but anything that can replace the separate `.batch_with` for each clusterable should be a win.

[the remark this answers is not in the document]
Comment (aelkiss) [reply lib/clustering/reclusterer.rb:58]: Yes, that's a good point.

**Pull request merged**